### PR TITLE
python3Packages.type-infer: init at 0.0.15

### PR DESCRIPTION
**Pull request by mbalatsko**
## Description of changes

- Fixed `nltk-data` output location
- Initialized `python3Packages.type-infer`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Dependency of https://github.com/NixOS/nixpkgs/issues/247089

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 256148` on x86_64-linux:

```
2 packages built:
python310Packages.type-infer python311Packages.type-infer
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg python310Packages.type-infer python311Packages.type-infer

File: pkgs/development/python-modules/type-infer/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    toml
+  ];
+
+  buildEnv = with lib; [
+    "NLTK_DATA=${testNltkData}"
+  ];
+
+  # pythonImportsCheck is not used as the package requires NLTK_DATA to be be set, or it starts downloading data
+  # There is no way to set environment variables for pythonImportsCheck
+  # without tests I want to at least check that the package can be imported
```
Comment (happysalada) [inline pkgs/development/python-modules/type-infer/default.nix:62]: what happens if you set `env.NLTK_DATA` for the entire derivation, does it still fail ?

**Comment by happysalada** [comment]
one single comment, appart from that this looks good.

File: pkgs/development/python-modules/type-infer/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/development/python-modules/type-infer/default.nix:62]: also you can do an export in a preCheck phase if that doesn't work.

Comment (mbalatsko) [reply pkgs/development/python-modules/type-infer/default.nix]: setting env var in `preCheck` would not help, but `env.NLTK_DATA` works fine. Thank you! Fixed!

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.type-infer python311Packages.type-infer

File: pkgs/development/python-modules/type-infer/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    toml
+  ];
+
+  # PyPI package does not include tests
+  doCheck = false;
+
+  # Package import requires NLTK data to be downloaded
+  # It is the only way to set NLTK_DATA environment variable,
+  # so that it is available in pythonImportsCheck
+  NLTK_DATA = testNltkData;
```
Comment (happysalada) [inline pkgs/development/python-modules/type-infer/default.nix:62]:
```suggestion
  env.NLTK_DATA = testNltkData;
```

**Review by happysalada — COMMENTED** [review]
I have one small nit, otherwise this looks good to me

File: pkgs/development/python-modules/type-infer/default.nix (hunk 2 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/type-infer/default.nix]: Done!

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.type-infer python311Packages.type-infer

**Comment by happysalada** [comment]
Thank you!

**Pull request merged**